### DISCUSSION TITLE
A doubt concerning enumeration(:)

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#11053] [Q&A] [answered]
Hi,

A class B where the following enumeration type is defined:
`replaceable type mcrChemicals = enumeration(:) ;`
gives the error: 
`[Redeclaration of enumeration 'mcrChemicals' is not a subtype of the redeclared element.]`
when I use it in a redeclaration:
```
extends B (redeclare type mcrChemicals = mcrChemicals1)
type mcrChemicals1 = enumeration(PM "Primary metabolite") "Macrochemical compounds";
```
However the problem does not appear if I write:
`extends B (redeclare type mcrChemicals = enumeration(PM))`

The error also disappears if the enumeration type is defined as
```
replaceable type mcrChemicals = chemicalsDummy;
type chemicalsDummy = enumeration(
    dummyCmp);
```
instead of enumeration(:) (independently of using mcrChemicals1 or enumeration(PM) in the extends clause).

Is it a bug? 

System: OMEdit 1.22 dev (16/8) and Windows 11.

Thanks four your help!

**Comment by perost** [comment, marked as answer]
Fixed in #11056.

The issue was this:
```Modelica
model A
  replaceable type E = enumeration(:);
end A;

model M
  extends A(redeclare type E = enumeration(a, b, c)); // Worked
  extends A(redeclare type E = E2); // Didn't work
  type E2 = enumeration(a, b, c);
end M;
```
I.e. the compiler didn't handle the case where the redeclare type wasn't an immediate enumeration but instead inheriting from one. With the fix both cases should now work.

**Reply by mpradovelasco** [reply]
Thanks a lot @perost ! I will test in my project in brief, using the new OMEdit 1.22-dev built that includes this fix.

**Reply by mpradovelasco** [reply]
Hi @perost, as commented I have tested successfully that fix on my project. Thanks again!
